### DISCUSSION TITLE
Watch multiple values and store data in object

### DISCUSSION THREAD

**Discussion by hdale94** [react-hook-form/react-hook-form#11492] [Q&A] [answered]
Hello,

I was looking through the docs, but could not find any info about this.

I know it's possible to watch multiple values by passing an array into the watch-method:

`const watchValues = watch(["date", "comment"]);`

Currently, to store these values in a object, I am referencing the indices of watchValues:

`const values = { date: watchValues[0], comment: watchValues[1] };`

This works fine, but I consider this not optimal when working with larger forms.

Is there a smarter way of achieving this without watching all values?

**Comment by jingeon27** [comment, marked as answer]
Try storing and utilizing the value using destructuring assignment. It will be a better experience than that.
```js
const [date,comment] = watch(["date", "comment"]);
const values = { date, comment };
```
